GraphicEx.pas

```diff
@@ -8883,8 +8883,14 @@ function TPNGGraphic.ReadImageProperties(const Memory: Pointer; Size: Int64; Ima
               end;
 
             Inc(Run, FHeader.Length + 4);
+            if (PAnsiChar(Run)>PAnsiChar(Memory)+Size) then //we have no IEND
+            //or got some trash instead of chunk size
+              Break;
+            //we could raise error here, too
             if IsChunk(IEND) then

```

Comment:
Wouldn't it make sense to first test for IEND (which is the normal case) and then do the sanity check only if that first check failed?